server/src/main/java/org/elasticsearch/action/admin/indices/resolve/ResolveIndexAction.java

```diff
@@ -469,20 +470,19 @@ protected void doExecute(Task task, Request request, final ActionListener<Respon
                 request.indices()
             );
             final OriginalIndices localIndices = remoteClusterIndices.remove(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
-            final Metadata metadata = clusterState.metadata();
             List<ResolvedIndex> indices = new ArrayList<>();
             List<ResolvedAlias> aliases = new ArrayList<>();
             List<ResolvedDataStream> dataStreams = new ArrayList<>();
             if (localIndices != null) {
+                assert request.includeDataStreams() : "expression resolver includes datastreams but request doesn't allow it";
```

Comment:
I think it's weird to assert the field here. Can we move it to the `Request` class itself? Or perhaps even make it a production runtime check within `Request.validate()`.

Reply:
The assert si close here to the code that relies on the assumption.
If I move it in the `Request` class, where the assumption is met, it loses it's power.
I've removed the assert and replaced it with a comment in [7dabfbc](https://github.com/elastic/elasticsearch/pull/92820/commits/7dabfbceeb26e3fb08fa259fdf33a76408d67f25)